### ISSUE TITLE
Add Icon Titles in Multiple Places

### ISSUE THREAD

**Issue by Sagar0-0** [kiwix/kiwix-android#3680]
**Describe the bug**

Some icons do not have a Title/Description on hold press.

**Expected behavior**

They should show their name/desc on hold.

**Screenshots**

![photo_6154636596016692419_y](https://github.com/kiwix/kiwix-android/assets/85388413/e82e4165-4488-4132-893e-3765e1d0126e)
![photo_6154636596016692420_y](https://github.com/kiwix/kiwix-android/assets/85388413/bb0d1975-a1a9-4349-86b5-02f59b2ad639)
![photo_6154636596016692421_y](https://github.com/kiwix/kiwix-android/assets/85388413/a26782e5-425a-43a0-9e76-5c2ccbb4473e)

**Environment**
- Version of Kiwix Android : 3.9.1
- Device : REALME GT
- OS version : 13

**Logs**

**Comment by kelson42** [comment]
@Sagar0-0 Do you know why the linting does not complain about this? Because that should be what happen in a first place right?

**Comment by Sagar0-0** [comment]
Before starting, I want you to know that Title was not present because we did not create this UI, we are using the androidx SearchView which is not implementing the titles by default. But, ig we can customize these and add the title programmatically.

**Comment by Sagar0-0** [comment]
We can create a kind of wrapper above SearchView and then use that only in all places instead of directly using Androidx.appcompat.SearchView in the menu Search item. Will that be fine, if I proceed with this approach?

**Comment by MohitMaliFtechiz** [comment]
@Sagar0-0 You should add the title on the existing SearchView's cancel image. We can get the cancel image of `SearchView` and we can add a title on it.

**Comment by Sagar0-0** [comment]
It have not worked, I have tried it already
1. I got the id of the view with androidx.appcompat.R.search_close_btn or something like that.
2. I added the title programmatically
3. Nothing changed

I also checked in the SearchView class there is only one close button or image.

**Comment by MohitMaliFtechiz** [comment]
"androidx.appcompat.R.search_close_btn" this is an image view so the title will not work for here, have you pointing to `contentDescripton`?

Have you tried to add the `tooltipText` on it?

**Comment by Sagar0-0** [comment]
> "androidx.appcompat.R.search_close_btn" this is an image view so the title will not work for here, have you pointing to `contentDescripton`?
Yes 
I've tried toolTip, and even tried passing the new Image but it was also not showing, so ig the Id could be wrong, but It was not.

**Comment by MohitMaliFtechiz** [comment]
Try with this code this is working.

```kotlin
 val closeImage =
            (searchItem.actionView as SearchView).findViewById<ImageView>(androidx.appcompat.R.id.search_close_btn)
          if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            closeImage?.tooltipText = "Clear query" // move this in string.xml file
          } else {
            closeImage?.contentDescription = "Clear query"
          }
```

**Comment by Sagar0-0** [comment]
![photo_6185854561512699952_y](https://github.com/kiwix/kiwix-android/assets/85388413/be1b0bf1-4b7e-4372-b343-fe681e8dca8f)
Yes, it is working, I was checking for the wrong view in the application. Dumb me :}
![photo_6185854561512699951_y](https://github.com/kiwix/kiwix-android/assets/85388413/10c7dd35-1686-4632-854d-289c2e0c6b7f)
Do you know the Id of this back button?

**Comment by Sagar0-0** [comment]
Also, we should create a common function or anything, which will add this description automatically while using SearchView. It might be not a good idea to use these everywhere.

**Comment by MohitMaliFtechiz** [comment]
> Do you know the ID of this back button?

It is a default toolbar back button.

> Also, we should create a common function or anything, which will add this description automatically while using SearchView. It might be not a good idea to use these everywhere.

Agreed.

**Comment by Sagar0-0** [comment]
> Agreed.

Then we are back to the same point. Should we add a SearchView wrapper or there can be other better ways?

**Comment by MohitMaliFtechiz** [comment]
What is your proposal for a wrapper?

**Comment by Sagar0-0** [comment]
> It is a default toolbar back button.

`activity.supportActionBar?.setHomeActionContentDescription(R.string.pref_back_to_top)` is not working

**Comment by Sagar0-0** [comment]
> What is your proposal for a wrapper? How you gonna implement this?

Something like this, but not completely sure
`<SearchView
    android:layout_width="wrap_content"
    android:layout_height="match_parent"
    android:searchIcon="@drawable/ic_search_white_24dp"
    android:closeIcon="@drawable/ic_clear_white_24dp" />`

**Comment by MohitMaliFtechiz** [comment]
I have not tested it why it is this not working. But it should.

> Something like this, but not completely sure
<SearchView android:layout_width="wrap_content" android:layout_height="match_parent" android:searchIcon="@drawable/ic_search_white_24dp" android:closeIcon="@drawable/ic_clear_white_24dp" />

Looks interesting go ahead!!

**Comment by MohitMaliFtechiz** [comment]
@Sagar0-0, @kelson42 This issue is a subpart of https://github.com/kiwix/kiwix-android/issues/3700, https://github.com/kiwix/kiwix-android/issues/3701 and this issue will automatically fixed by https://github.com/kiwix/kiwix-android/pull/3706 and https://github.com/kiwix/kiwix-android/pull/3707.

* For the close icon of search, we have placed a fix in https://github.com/kiwix/kiwix-android/pull/3706, see the PR description.
  Example:- 

![clearquery1](https://github.com/kiwix/kiwix-android/assets/34593983/12c7fbc4-1f2d-4b7b-a872-11750c876294)

* For the toolbar's back button, we have placed a fix in https://github.com/kiwix/kiwix-android/pull/3707.
  Example:- 
![Screenshot_20240216-123421](https://github.com/kiwix/kiwix-android/assets/34593983/05ad32fa-d529-4b2b-81a3-082d76bcc201)


> activity.supportActionBar?.setHomeActionContentDescription(R.string.pref_back_to_top) is not working(testing in LanguageFragment). Do you have any idea, why?

@Sagar0-0 `contentDescripton` is working for the back button, but it will not show if the user long clicks on that. See https://developer.android.com/develop/ui/views/components/tooltips for more details. There is no direct way to do it as there is no direct method available to get the back button from the toolbar, I have achieved it in https://github.com/kiwix/kiwix-android/pull/3707.

**Comment by kelson42** [comment]
All fixes implemented and merged!
